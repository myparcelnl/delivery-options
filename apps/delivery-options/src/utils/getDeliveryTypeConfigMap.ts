import {
  ALLOW_DELIVERY_OPTIONS,
  ALLOW_EVENING_DELIVERY,
  ALLOW_MORNING_DELIVERY,
  ALLOW_PICKUP_LOCATIONS,
  type CarrierSetting,
  type SupportedDeliveryTypeName,
} from '@myparcel-do/shared';
import {DeliveryTypeName} from '@myparcel/constants';

export const getDeliveryTypeConfigMap = (): Record<SupportedDeliveryTypeName, CarrierSetting> => ({
  [DeliveryTypeName.Standard]: ALLOW_DELIVERY_OPTIONS,
  [DeliveryTypeName.Evening]: ALLOW_EVENING_DELIVERY,
  [DeliveryTypeName.Morning]: ALLOW_MORNING_DELIVERY,
  [DeliveryTypeName.Pickup]: ALLOW_PICKUP_LOCATIONS,
});
