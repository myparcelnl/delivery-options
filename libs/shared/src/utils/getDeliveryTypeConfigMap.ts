import {CarrierSetting, CustomDeliveryType, type SupportedDeliveryTypeName} from '@myparcel-do/shared';
import {DeliveryTypeName} from '@myparcel/constants';

export const getDeliveryTypeConfigMap = (): Record<SupportedDeliveryTypeName, CarrierSetting> => ({
  [DeliveryTypeName.Standard]: CarrierSetting.AllowDeliveryOptions,
  [DeliveryTypeName.Evening]: CarrierSetting.AllowEveningDelivery,
  [DeliveryTypeName.Morning]: CarrierSetting.AllowMorningDelivery,
  [DeliveryTypeName.Pickup]: CarrierSetting.AllowPickupLocations,
  [CustomDeliveryType.SameDay]: CarrierSetting.AllowSameDayDelivery,
});
