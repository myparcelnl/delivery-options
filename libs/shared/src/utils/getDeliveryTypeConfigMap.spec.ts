import {describe, expect, it} from 'vitest';
import {getDeliveryTypeConfigMap} from './getDeliveryTypeConfigMap';

describe('getDeliveryTypeConfigMap', () => {
  it('should return the correct config map', () => {
    expect(getDeliveryTypeConfigMap()).toMatchSnapshot();
  });
});
