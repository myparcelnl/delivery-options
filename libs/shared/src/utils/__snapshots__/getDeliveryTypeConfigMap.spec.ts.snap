// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDeliveryTypeConfigMap > should return the correct config map 1`] = `
{
  "evening": "allowEveningDelivery",
  "monday": "allowMondayDelivery",
  "morning": "allowMorningDelivery",
  "pickup": "allowPickupLocations",
  "same_day": "allowSameDayDelivery",
  "saturday": "allowSaturdayDelivery",
  "standard": "allowDeliveryOptions",
}
`;
